eclipse/che
# Jwtproxy is the new default for the secure servers

### What does this PR do?

This changes the default implementation of the secure servers authentication to "jwtproxy". The previous default was a mere passthrough, which offered no che-provided authentication at all.

This PR also consolidates the secure server behavior of plugins and other workspace components. Before, the plugins and editors were authenticated using jwtproxy regardless of how what the `che.server.secure_exposer` said.

### What issues does this PR fix or reference?
#17852

### How to test this PR?
1. Deploy a fresh che server built from this branch.
1. Deploy a new workspace using the below devfile.
1. Make sure that endpoint works and is correctly authenticated
    1. Start the quarkus application in devmode
    1. Open web tools in the browser
    1. Click on the endpoint link
    1. In the network tab of the web tools check that the auth flow has happened and that the final request to the endpoint page has its authorization cookie set.

```yaml
metadata:
  name: test-for-17852
projects:
  - name: quarkus-quickstarts
    source:
      location: 'https://github.com/quarkusio/quarkus-quickstarts.git'
      type: git
      sparseCheckoutDir: getting-started
components:
  - id: redhat/quarkus-java11/latest
    type: chePlugin
  - mountSources: true
    endpoints:
      - name: hello-greeting-endpoint
        port: 8080
        attributes:
          cookiesAuthEnabled: 'true'
          path: /
          secure: 'true'
    memoryLimit: 4927M
    type: dockerimage
    volumes:
      - name: m2
        containerPath: /home/user/.m2
    alias: centos-quarkus-maven
    image: 'quay.io/eclipse/che-quarkus:nightly'
    env:
      - value: >-
          -XX:MaxRAMPercentage=50.0 -XX:+UseParallelGC -XX:MinHeapFreeRatio=10
          -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4
          -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true
          -Xms20m -Djava.security.egd=file:/dev/./urandom -Duser.home=/home/user
        name: JAVA_OPTS
      - value: $(JAVA_OPTS)
        name: MAVEN_OPTS
apiVersion: 1.0.0
commands:
  - name: Start Development mode (Hot reload + debug)
    actions:
      - workdir: '${CHE_PROJECTS_ROOT}/quarkus-quickstarts/getting-started'
        type: exec
        command: './mvnw compile quarkus:dev'
        component: centos-quarkus-maven
```
 
### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
